src/app/pages/profile/transaction-information/transaction-information.component.spec.ts

```diff
@@ -0,0 +1,156 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { DebugElement } from '@angular/core';
+import { HttpClientModule } from '@angular/common/http';
+import { of } from 'rxjs';
+import { By } from '@angular/platform-browser';
+
+import { AppModule } from './../../../app.module';
+import { resetSpies, transactionServiceSpy } from './../../../helpers/spies';
+import { TransactionInformationComponent } from './transaction-information.component';
+import { TransactionInformationService } from 'src/app/services/profile/transaction-information/transaction-information.service';
+import { EmptyTransactionComponent } from './empty-transaction/empty-transaction.component';
+
+describe('TransactionInformationComponent', () => {
+  let component: TransactionInformationComponent;
+  let fixture: ComponentFixture<TransactionInformationComponent>;
+  let debugElement: DebugElement;
+  const url =
+    'https://landville-backend-web-api.herokuapp.com/api/v1/transactions/';
+
+  const Mockresponse = {
+    data: {
+      transactions: [
+        {
+          title: 'Plazza de ciel',
+          buyer: 'jeandedieuam@gmail.com',
+          price: 500.0,
+          total_amount_paid: 500.0,
+          balance: 0.0,
+          deposits: [
+            {
+              date: '2019-08-20T16:47:02.723002Z',
+              amount: 500.0
+            }
+          ],
+          percentage_completion: '100.00',
+          image_main:
+            'http://res.cloudinary.com/landville/image/upload/v1566319487/gxeuj5s7nxi3ulxoff7g.jpg',
+          address: {
+            City: 'Bukavu',
+            State: 'Congo',
+            Street: 'Nyawera'
+          }
+        }
+      ]
+    }
+  };
+  beforeAll(() => resetSpies([transactionServiceSpy]));
+  afterEach(() => resetSpies([transactionServiceSpy]));
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        TransactionInformationComponent,
+        EmptyTransactionComponent
+      ],
+      imports: [AppModule, HttpClientModule],
+      providers: [
+        {
+          provide: TransactionInformationService,
+          useValue: transactionServiceSpy
+        }
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TransactionInformationComponent);
+    component = fixture.componentInstance;
+    transactionServiceSpy.getTransactions.and.returnValue(of(Mockresponse));
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should call getTransactions method', () => {
+    const url = 'http://api/v1/transactions';
```

Comment:
@Jaman-dedy , I believe we can have this `url` at the very top of the file so that it can be referenced on each test function instead of having `const url = ...` variable for each `it block`

Reply:
Thanks @Tevinthuku , I worked on that